addon/templates/components/frost-modal-dialog.hbs

```diff
@@ -59,7 +59,16 @@
   {{! END-SNIPPET }}
 
   {{#if params.footer}}
-    <div class="frost-modal-dialog-footer-text">{{params.footer}}</div>
+    {{#if (type-of params.footer 'object')}}
+      <div class="frost-modal-dialog-footer-object"
+        data-test={{hook (concat hook '-object')}}>
+        {{component params.footer}}
```

Comment:
@pbosse looks like the type-of check and the internal component helper are unnecessary (sorry) - let's go ahead and remove those and change the class of this div to `frost-modal-dialog-footer-content` (generic away from `text`)

Otherwise the PR is fine (docs and test updates) - the documentation update should be enough to clarify this for any future consumers